api/src/main/proto/stellarstation/api/v1/groundstation/groundstation.proto

```diff
@@ -159,6 +159,23 @@ message UnavailabilityWindow {
 
   // End time of the unavailability window.
   google.protobuf.Timestamp end_time = 3;
+
+  // Indicates whether this window is part of a recurring set.
+  enum RecurringPattern {
+    // This is a one-off window that does not recur.
+    NONE = 0;
+
+    // The window recurs daily at the same time.
+    DAILY = 3;
+
+    // The window recurs weekly on the same day.
+    WEEKLY = 1;
+  }
+
+  // Details of the recurrence pattern.
+  //
+  // Note that 'window_id' is the same for all instances of a recurring window.
```

Comment:
Yeah, I guess that's true. The alternatives are to do something special with the ID (bit of a hack) or deprecate 'start_time' and 'end_time' and have a repeating 'range' instead.

Or start and end can be the first occurrence with a repeated field for the recurring events (so that first one could be before the range requested).